### ISSUE TITLE
'table' ignores custom wayHandler, 'route' doesn't

### ISSUE THREAD

**Issue by Doidel** [Project-OSRM/osrm-backend#6587]
We run `osrm-extract` with a custom profile where we have some additional blocked roads.
When we calculate routes with `route` it works as expected and doesn't route through these nodes. Yet when I query the same source & destination with `table` the roadblock is ignored.

For reference, here's the wayhandler we use:
```lua
function WayHandlers.roadblocks_forcedWays(profile,way,result,data)
  local wayId = way:id()
  for line in io.lines("/data/roadblocks_forcedways.csv") do
    local wayIdToEdit = tonumber(line)

    if (wayIdToEdit < 0) and (-wayIdToEdit == wayId) -- negativ way id -> road block
    then
      print('blocked way = ', -wayIdToEdit)
      result.forward_speed = 0
      result.backward_speed = 0
      result.duration = 999999999
      result.weight = 999999999
      result.forward_mode = mode.inaccessible
      result.backward_mode = mode.inaccessible
      result.forward_restricted = true
      result.backward_restricted = true
      result.is_startpoint = false
      return
    elseif (wayIdToEdit > 0) and (wayIdToEdit == wayId) -- positiv way id -> forced way
    then
      print('forced way = ', wayIdToEdit)
      result.forward_mode = mode.driving
      result.backward_mode = mode.driving
      result.forward_restricted = false
      result.backward_restricted = false
      result.is_startpoint = true
    end
  end
end
```
This is our `table` query (which routes through blocks):

`table/v1/driving/8.52857,47.28761;8.51817,47.27599;8.52857,47.28761;8.51817,47.27599.json?sources=0;1&destinations=2;3&annotations=duration,distance&fallback_coordinate=input&generate_hints=true&snapping=default&skip_waypoints=false`

This is our `route` query, which correctly avoids roadblocks:
`route/v1/driving/8.52928,47.2878;8.51813,47.27573.json?alternatives=false&steps=true&annotations=false&geometries=geojson&overview=false&continue_straight=default&generate_hints=true&snapping=default&skip_waypoints=false`

And for reference our extraction command:
`osrm-extract -p /our/custom/profile.lua --parse-conditional-restrictions --generate-edge-lookup ch-fl.osm.pbf`

I'd be grateful for any hints on where I should even start looking.
Thanks!

**Comment by SiarheiFedartsou** [comment]
AFAIR `table` API doesn't return any route information(like geometry or node ids): just duration and distance. I am curious, how do you know that OSRM actually routes through "blocked" way?

**Comment by Doidel** [comment]
That's true, yes. The reason I'm fairly certain though is because the differences are so staggering. Here's a visualization where we try to go from customer 9 to customer 10 (customer 9 is slightly above 18):
![image](https://user-images.githubusercontent.com/833039/229132686-f3a63bd9-af1a-4dd9-b6dc-a5074bcfbb11.png)
The blue path is the route with `route`, which takes 1360s. The duration of the distance matrix is 317s (both back and forth).

**Comment by Doidel** [comment]
This btw also happens when I just query this one distance from **9** to **10** with the `table` API. I get
[ 0   317 ]
[317    0 ]
instead of the 1360 returned by `route`

**Comment by Doidel** [comment]
It was indeed an issue on our side. We routed to slightly different coordinates, and by chance this small difference resulted in different snapping outcomes (to the east and west of the blocked road).

Thanks for the answer nonetheless, it helped me to double down on our own code.
